### ISSUE TITLE
Resource coordination.k8s.io/Lease strimzi-cluster-operator is excluded in the settings

### ISSUE THREAD

**Issue by teimyBr** [argoproj/argo-cd#12243]
Checklist:

* [ x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [ x] I've included steps to reproduce the bug.
* [ x] I've pasted the output of `argocd version`.

**Describe the bug**

ExcludedResourceWarning
Resource coordination.k8s.io/Lease strimzi-cluster-operator is excluded in the settings

Since we update Strimzi to 0.33 Argo CD show this Warning (see above) in the Application

**To Reproduce**

Install Strimzi 0.33 and Argo CD 2.5.9

**Expected behavior**

Argo CD does not log this warning

**Version**

Argo CD 2.5.9

**Logs**

ExcludedResourceWarning
Resource coordination.k8s.io/Lease strimzi-cluster-operator is excluded in the settings
```
Paste any relevant application logs here.
```

**Comment by mananpreetsingh** [comment]
I also had the same issue.

Cause of issues on Kafka's side as well. I found while upgrading to 0.33 manually on a different cluster.
With this version 0.33 it needs to have specific labels to all the resources it creates. one of the resources that does not has proper labels in the past is Leases `lease.coordination.k8s.io "strimzi-cluster-operator"` 

The manual fix if deployed using helm is: to uninstall strimzi Kafka and then delete this lease resource manually. 

But Argo uses kubectl apply in the backend to apply so no sure how it will handle this. 

In argocd I also t ried to uninstall stmrizi kafka and then reinstall but that did not help, `lease.coordination.k8s.io "strimzi-cluster-operator"` always comes without proper labels, which  triggers above warning.

**Comment by alfsch** [comment]
> I also had the same issue.
> 
> Cause of issues on Kafka's side as well. I found while upgrading to 0.33 manually on a different cluster. With this version 0.33 it needs to have specific labels to all the resources it creates. one of the resources that does not has proper labels in the past is Leases `lease.coordination.k8s.io "strimzi-cluster-operator"`
> 
> The manual fix if deployed using helm is: to uninstall strimzi Kafka and then delete this lease resource manually.
> 
> But Argo uses kubectl apply in the backend to apply so no sure how it will handle this.
> 
> In argocd I also t ried to uninstall stmrizi kafka and then reinstall but that did not help, `lease.coordination.k8s.io "strimzi-cluster-operator"` always comes without proper labels, which triggers above warning.

@mananpreetsingh what are the proper labels from your point of view?

**Comment by jaideepr97** [comment]
hi @teimyBr thanks for raising this
would you mind elaborating on steps to reproduce so someone can try it? If you have a sample application to use that would be helpful as well

**Comment by alfsch** [comment]
Simply install the strimzi-kafka-operator helm chart in version 0.33.0

```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: strimzi-operator
  namespace: argocd
spec:
  project: default
  source:
    chart: strimzi-kafka-operator
    repoURL: https://strimzi.io/charts
    targetRevision: 0.33.0
    helm:
      releaseName: strimzi-operator
  destination:
    server: "https://kubernetes.default.svc"
    namespace: strimzi-operator
```

Hope this helps.

**Comment by mananpreetsingh** [comment]
@alfsch 

if you install using `helm upgrade` command you will see proper label.  When I try to upgrade using `helm upgrade` command from 0.32 to 0.33 version I got following error, which mean atleast these labels are needed:

`Error: UPGRADE FAILED: rendered manifests contain a resource that already exists. Unable to continue with update: Lease "strimzi-cluster-operator" in namespace "strimzi-kafka" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "strimzi-kafka"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "strimzi-kafka"`

So from this error these labels are needed:

**labels that are needed:**

`app.kubernetes.io/managed-by:  Helm`

**annotations that are needed:**
```
meta.helm.sh/release-name: strimzi-kafka
meta.helm.sh/release-namespace: strimzi-kafka
```

But with proper installation using helm command, I do see other labels as well:
```
app: strimzi
chart: strimzi-kafka-operator-0.33.0
component: lease
heritage: Helm
release: strimzi-kafka
```

Basically Lease has the same labels as deployment now. By saying since argo do not use helm in the backend so wonder if helm labels will be needed or only rest of it.

**Comment by teimyBr** [comment]
@jaideepr97  
Yes install Argo CD 2.5.9

with 
```
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: strimzi-operator
  namespace: argocd
spec:
  project: default
  source:
    chart: strimzi-kafka-operator
    repoURL: https://strimzi.io/charts
    targetRevision: 0.33.0
    helm:
      releaseName: strimzi-operator
  destination:
    server: "https://kubernetes.default.svc"
    namespace: strimzi-operator
```

then you see

ExcludedResourceWarning
Resource coordination.k8s.io/Lease strimzi-cluster-operator is excluded in the settings

**Comment by teimyBr** [comment]
https://github.com/strimzi/strimzi-kafka-operator/releases/tag/0.33.1

Strimzi removed this coordination.k8s.io/Lease and this solved this.

**Comment by teimyBr** [comment]
#https://github.com/strimzi/strimzi-kafka-operator/pull/8021
